test/helpers/position_assertions.cc

```diff
@@ -1129,4 +1130,119 @@ void ApplyCodeActionAssertion::check(const UnorderedMap<std::string, std::shared
     }
 }
 
+SymbolSearchAssertion::SymbolSearchAssertion(string_view filename, unique_ptr<Range> &range, int assertionLine,
+                                             string_view query)
+    : RangeAssertion(filename, range, assertionLine), query(query) {}
+
+shared_ptr<SymbolSearchAssertion> SymbolSearchAssertion::make(string_view filename, unique_ptr<Range> &range,
+                                                              int assertionLine, string_view assertionContents,
+                                                              string_view assertionType) {
+    static const regex contentsRegex(R"(^\s*\"([^\"]+)\"\s*$)");
+
+    smatch matches;
+    string assertionContentsString = string(assertionContents);
+    if (regex_search(assertionContentsString, matches, contentsRegex)) {
+        auto query = matches[1].str();
+        return make_shared<SymbolSearchAssertion>(filename, range, assertionLine, query);
+    }
+
+    ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1) << fmt::format(
+        "Improperly formatted assertion. Expected 'symbol-search: \"<query>\"'. Found '{}'", assertionContents);
+
+    return nullptr;
```

Comment:
You should probably `Exception::raise` here to throw an exception to abort the test.